+sw_tests/+unit_tests/unittest_spinw_genmagstr.m

```diff
@@ -0,0 +1,540 @@
+             % S with helical must be real
+             {{'mode', 'helical', 'S', [1.5; 0; 1.5i]}, 'spinw:genmagstr:WrongInput'}; ...
+             % Rotate mode must first initialise a magnetic structure
+             {{'mode', 'rotate'}, 'spinw:genmagstr:WrongInput'}
+             };
+         rotate_input_errors = { ...
+             % varargin, identifier
+             % rotation angle must be real (previously phi=i had special
+             % behaviour)
+             {{'mode', 'rotate', 'phi', i}, 'spinw:genmagstr:ComplexPhi'}
+             % If no angle is supplied to rotate, the rotation axis is set
+             % orthogonal to both the first spin and n. If the first spin
+             % and n are parallel, this should therefore cause an error
+             {{'mode', 'rotate'}, 'spinw:genmagstr:InvalidRotation'}; ...
+             };
+    end
+    methods (TestMethodSetup)
+        function setup_chain_model(testCase)
+            % Create a simple FM 1D chain model
+            testCase.swobj = spinw();
+            testCase.swobj.genlattice('lat_const', [3 8 8], 'angled', [90 90 90]);
+            testCase.swobj.addatom('r', [0 0 0],'S', 1);
+        end
+        function setup_tri_model(testCase)
+            % Create a simple triangular lattice model
+            testCase.swobj_tri = spinw;
+            testCase.swobj_tri.genlattice('lat_const', [4 4 6], 'angled', [90 90 120]);
+            testCase.swobj_tri.addatom('r', [0 0 0], 'S', 3/2);
+        end
+    end
+
+    methods (Test)
+        function test_invalid_fm_chain_input_raises_error(testCase, fm_chain_input_errors)
+            varargin = fm_chain_input_errors{1};
+            identifier = fm_chain_input_errors{2};
+            testCase.verifyError(...
+                @() testCase.swobj.genmagstr(varargin{:}), ...
+                identifier)
+        end
+        function test_invalid_rotate_input_raises_error(testCase, rotate_input_errors)
+            varargin = rotate_input_errors{1};
+            identifier = rotate_input_errors{2};
+            swobj = copy(testCase.swobj);
+            swobj.genmagstr('mode', 'direct', 'S', [0; 0; 1]);
+            testCase.verifyError(...
+                @() swobj.genmagstr(varargin{:}), ...
+                identifier)
+        end
+        function test_no_magnetic_atoms_raises_error(testCase)
+            swobj = spinw;
+            testCase.verifyError(...
+                @() swobj.genmagstr(), ...
+                'spinw:genmagstr:NoMagAtom')
+        end
+        function test_tile_too_few_S_raises_error(testCase)
+            swobj = copy(testCase.swobj);
+            swobj.addatom('r', [0.5 0.5 0], 'S', 1);
+            testCase.verifyError(...
+                @() swobj.genmagstr('mode', 'tile', 'S', [0; 1; 1]), ...
+                'spinw:genmagstr:WrongInput')
+        end
+        function test_helical_spin_size_incomm_with_nExt_warns(testCase)
```

Comment:
This test does throw the error `'spinw:genmagstr:UCExtNonSuff'` but it doesn't look like there is a test with `epsilon` explicitly provided that throws the same error (i.e. that confirms espilon is being used correctly. Something along the line of?
```
delta = 1e-2;
k = [(delta+1)/3, 0, 0];
nExt = [3,1,1];
testCase.verifyWarning(...
    @() testCase.swobj.genmagstr('mode', 'helical', 'S', [1; 0; 0], ...
                                 'k', k, 'epsilon', 1.01*delta, ...
                                 'nExt', nExt), ...
    'spinw:genmagstr:UCExtNonSuff');
```

Reply:
Done! I had to make the input epsilon smaller that the default epsilon rather than larger otherwise due to the `>` it would pass anyway

Reply:
Thanks yes that was a typo - I copied the wrong `genmagstr` call into github!